arraycontext/container/traversal.py

```diff
@@ -533,20 +534,22 @@ def freeze(
 
     See :meth:`ArrayContext.thaw`.
     """
-    try:
-        iterable = serialize_container(ary)
-    except NotAnArrayContainerError:
-        if actx is None:
-            raise TypeError(
-                    f"cannot freeze arrays of type {type(ary).__name__} "
-                    "when actx is not supplied. Try calling actx.freeze "
-                    "directly or supplying an array context")
-        else:
-            return actx.freeze(ary)
+    if actx is None:
```

Comment:
We should be able to make `freeze` not `@singledispatch` here then, right?

Reply:
Yep, removed! See https://github.com/inducer/arraycontext/compare/d373c46d46254bc539140152503c834a4fd3d441..cbd3adea6b48be9ca3d384106ba60125734e3e7f